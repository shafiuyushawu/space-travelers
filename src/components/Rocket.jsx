import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../redux/rockets/rocketsSlice';

const Rocket = ({ rocket }) => {
  const dispatch = useDispatch();

  const handleReservation = () => {
    dispatch(reserveRocket(rocket.id));
  };

  return (
    <div className="rocket">
      <img src={rocket.flickr_images[0]} alt={rocket.rocket_name} />
      <div className="rocket-detail">
        <h2>{rocket.rocket_name}</h2>
        <p>
          {rocket.reserved && (
          <Badge bg="info">
            Reserved
          </Badge>
          )}
          {rocket.description}
        </p>
        {
          rocket.reserved
            ? <Button variant="outline-secondary" onClick={handleReservation}>Reserve Rocket</Button>
            : <Button variant="primary" onClick={handleReservation}>Reserve Rocket</Button>
        }
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string,
    rocket_name: PropTypes.string,
    flickr_images: PropTypes.arrayOf(PropTypes.string),
    description: PropTypes.string,
    reserved: PropTypes.bool,
  }).isRequired,
};

export default Rocket;
