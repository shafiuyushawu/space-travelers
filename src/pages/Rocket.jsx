import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';

const Rocket = ({ rocket }) => (
  <div className="rocket">
    <img src={rocket.flickr_images[0]} alt={rocket.rocket_name} />
    <h2>{rocket.rocket_name}</h2>
    <p>{rocket.description}</p>
    <Button variant="primary">Reserve Rocket</Button>
  </div>
);

Rocket.propTypes = {
  rocket: PropTypes.shape({
    rocket_name: PropTypes.string,
    flickr_images: PropTypes.arrayOf(PropTypes.string),
    description: PropTypes.string,
  }).isRequired,
};

export default Rocket;
