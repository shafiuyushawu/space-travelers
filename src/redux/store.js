import { configureStore } from '@reduxjs/toolkit';
import missionSlice from './missions/missionSlice';
import rocketReducer from './rockets/rocketsSlice';

const store = configureStore({
  missions: missionSlice,
  rockets: rocketReducer,
});
export default store;
