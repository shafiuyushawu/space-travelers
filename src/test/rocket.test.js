import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Rocket from '../components/Rocket';

afterEach(cleanup)
describe('Testing rocket components', () => {
    test('Generating the snapshots for rocket component', () => {
        const rocket = {
            id: '123',
            rocket_name: 'Test Rocket',
            flickr_images: ['image1.jpg', 'image2.jpg'],
            description: 'Test rocket description',
            reserved: false,
        };
        const { container } = render(
            <Provider store={store}>
                <Rocket rocket={rocket} />
            </Provider>,
        );
        expect(container).toMatchSnapshot();
    });
});