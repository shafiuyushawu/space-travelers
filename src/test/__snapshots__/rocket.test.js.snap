// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rocket components Generating the snapshots for rocket component 1`] = `
<div>
  <div
    class="rocket"
  >
    <img
      alt="Test Rocket"
      src="image1.jpg"
    />
    <div
      class="rocket-detail"
    >
      <h2>
        Test Rocket
      </h2>
      <p>
        Test rocket description
      </p>
      <button
        class="btn btn-primary"
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
